src/bindings/python/flux/util.py

```diff
@@ -147,6 +147,35 @@ def _format_action_invocation(self, action):
     return lambda prog: FluxHelpFormatter(prog, max_help_position=argwidth)
 
 
+def set_treedict(in_dict, key, val):
+    """
+    set_treedict(d, "a.b.c", 42) is like d[a][b][c] = 42
+    but levels are created on demand.
+    """
+    path = key.split(".", 1)
+    if len(path) == 2:
+        set_treedict(in_dict.setdefault(path[0], {}), path[1], val)
+    else:
+        in_dict[key] = val
+
+
+class TreedictAction(argparse.Action):
+    """argparse action that returns a diction from a set of "treedict" values"""
```

Comment:
did you mean "dictionary"?

Reply:
Yeah, thanks for catching that!

Actually the documentation of that function is pretty weak, so I've fixed the error you found and also expanded the documentation.